src/services/ApplicationsRegistry.cpp

```diff
@@ -5,9 +5,12 @@
 
 ApplicationsRegistry::ApplicationsRegistry(QStringList appDirs)
     : _appDirs(std::move(appDirs))
+    , _appsDBHelper(_appsDBHelper->getInstance())
```

Comment:
The ApplicationRegistry triggers signals when a change occurs those can be used to persist the changes on the cache. Therefore, there is no need to have the db as a dependency of the registry.